src/rates.py

```diff
@@ -1,9 +1,41 @@
-import numpy as np
 import pandas as pd
+import numpy as np
 import plotly.graph_objects as go
 from qablet.base.utils import Discounter
 
+# Function to fetch the rates dynamically from the Treasury website using a CSV URL
+def fetch_treasury_rates():
+    csv_url = "https://home.treasury.gov/resource-center/data-chart-center/interest-rates/daily-treasury-rates.csv/2024/all?type=daily_treasury_yield_curve&field_tdr_date_value=2024&page&_format=csv"
+    df = pd.read_csv(csv_url)
+    return df
+
+# Helper function to get the rates for a specific pricing date
+def get_rates_for_date(pricing_datetime):
+    df = fetch_treasury_rates()
+    # Convert date column to datetime and filter for the closest available date
+    df['Date'] = pd.to_datetime(df['Date'])
+    closest_date = df.iloc[(df['Date'] - pricing_datetime).abs().argsort()[:1]]
+    return closest_date
 
+# Function to convert the Treasury yield curve rates into a usable format for pricing
+def treasury_rates_to_rate_data(df_row):
```

Comment:
define a constant like, 
TREASURY_KNOTS = [
    [1/12, "1 Mo"],
    [2/12, "2 Mo"],
    ...
]

and use it here. the same constant will be usefull in some key rate risk report we will do later.